packages/good-design/src/apps/ubi/context/ClaimContext.tsx

```diff
@@ -0,0 +1,233 @@
+  onNews,
+  onSendTx,
+  onSuccess
+}) => {
+  const { account, chainId, library, switchNetwork } = useEthers();
+  const [refreshRate, setRefreshRate] = useState<QueryParams["refresh"]>(4);
+  const [preClaimPools, setClaimPools] = useState<any[]>([]);
+  const [postClaimPools, setPostClaimPools] = useState<any[]>([]);
+  const activeChain = SupportedChains[chainId as number] ?? "CELO";
+  const endpoints = explorerEndPoints[activeChain as keyof typeof SupportedChains];
+  const [error, setError] = useState<string | undefined>(undefined);
+  const claimedAlt = useClaimedAlt(chainId);
+
+  const [txDetails, setTxDetails] = useState({ transaction: undefined, isOpen: false });
+
+  const formattedTransactionList = useFormatClaimTransactions(
+    postClaimPools.length > 0 ? postClaimPools : preClaimPools.length > 0 ? preClaimPools : [],
+    chainId,
+    account ?? ""
+  );
+
+  const claimDetails = useClaim(refreshRate);
+  const { poolsDetails, loading, fetchPools } = useGetMemberUBIPools();
+
+  const { poolContracts, setContract, setPoolContracts, transactionState, claimFlowStatus } =
+    useMultiClaim(preClaimPools);
+  const { errorMessage } = transactionState?.state ?? {};
+
+  const onTxDetails = useCallback(
+    (transaction: any) => {
+      console.log("transactionTxDetails -->", { transaction });
+      setTxDetails({ transaction, isOpen: true });
+    },
+    [txDetails]
+  );
+
+  const onClaimFailed = useCallback(async () => {
+    setError(errorMessage ?? /*i18n*/ "An unknown error occurred while claiming");
+
+    if (claimFlowStatus.isClaimingDone) {
+      resetState();
+    }
+  }, [errorMessage, claimFlowStatus]);
+
+  const resetState = () => {
+    setRefreshRate("everyBlock");
+    setClaimPools([]);
+    setPostClaimPools([]);
+    setPoolContracts([]);
+    void fetchPools();
+
+    // because of awaiting on-chain data (eg. contract status / tx validation) and context switching (account/network)
+    // combined with the varying states of the claim process (pre-claim, post-claim, with/without additional pools),
+    // we can end up in a wrong or mixed up page-state.
+    // why we set a short lock to handle certain racing-conditions
+    void explorerPollLock.acquire("resetLock");
+
+    setTimeout(async () => {
+      void explorerPollLock.release("resetLock");
+      setRefreshRate(4);
+    }, 300);
+  };
+
+  const onClaim = useCallback(async () => {
+    // todo: handle onboard/upgrade flow?
+
+    // initializes a cycle of claim transactions for all contracts
+    setContract(poolContracts[0]);
+  }, [poolsDetails, onSendTx, poolContracts]);
+
+  const onClaimSuccess = useCallback(async () => {
+    transactionState?.resetState();
+    resetState();
+
+    // should handle what happens after all claims are done (eg. showing a next-task modal)
+    // if nothing is done, it will just silently finish
+    await onSuccess?.();
+  }, [transactionState, onSuccess]);
+
+  const switchChain = useCallback(() => {
```

Comment:
whats this?

Reply:
Which one, the switchChain or onSuccess?

Reply:
I assume you are referring to the switchChain implementation.

The comment 'to discuss' is related to something to discuss UI wise (forgot to discuss with @decentralauren)